### ISSUE TITLE
Failing to train a new generator on a single author custom non-english dataset

### ISSUE THREAD

**Issue by LinasEin** [EDM-Research/VATr-pp#4]
Hello,

Awesome work! Your pre-trained text generator on IAM dataset has really good quality. I am also interested in training a new generator that could work with other language (mostly latin characters with some special č,š,ž.. letters). Unfortunately, I face numerical stability issues.

Some background:
I have prepared a new LANG-32.pickle file with word level images (words and numbers) of single author style. 1638 images in total (1472 train,  166 test). Originally these images had background in them, which I removed using Otsu thresholding technique. See 
example bellow:
![unnamed1](https://github.com/user-attachments/assets/e3249e67-e9ef-4d0e-8a71-ad68cf45ee1e)

Training command:
```python  train.py --dataset "LANG" --english_words_path "file/custom-words.txt" --num-writers 1 --num-examples 1 -d-crop-size 64 128 --text-augment-strength 0.4```
I have also changed all LRs to be 1e-5 in misc.py.

Now the issue is with the training. Firstly, I constantly get:
```WARNING: alpha > 1000 or alpha < 0.0001 - alpha=7120.94287109375```
```WARNING: alpha > 1000 or alpha < 0.0001 - alpha=5.806722810919318e-11```
This leads me to assume that I am dealing with exploding/vanishing gradients issue here. In any case, I managed to reach around 1500 epochs, but now training constantly stops with loss OCR_fake becoming NaN. Once restarted it stops after another ~30 epochs with the same error.

On failure this gets printed in backward_G_only():
```
loss OCR fake:  tensor(nan, device='cuda:0', grad_fn=<MulBackward0>)  loss_G:  tensor(10.8988, device='cuda:0', grad_fn=<AddBackward0>)  words:  [b'tavasaF', b'Kupst\xc4\x99n\xc4\x97', b'1916-\xc4\x99-20', b'193\xc4\xaf/12/29', b'gaiv\xc5\xaan\xc4\x85b', b'mi\xc5\xa1rNnti', b'Nicloz\xc4\x85', b'e\xc5\xbe']
```

Things I tried include:
- Running pred_real=pred_real.log_softmax(2) and pred_fake_OCR=pred_fake_OCR.log_softmax(2) (on the real and fake ocr network output) so that log probabilities would sum up to 1.
- Adjusting text augmentation strength to 0.2.
- Turning off completely text augmentation and discriminator augmentation.

Generally, the model seems to be slowly learning, but just stops with loss_fake_ocr being NaN. Image from wandb:
<img width="1410" alt="image" src="https://github.com/user-attachments/assets/29327eab-ce79-402c-afb0-7537767f6187">

Perhaps you may have some ideas what else could I try?

Thanks!

**Comment by AlekseevArsenii** [comment]
Hello! Do you have any updates? Did you manage to fine-tune it on some other language?

**Comment by AlekseevArsenii** [comment]
Hello! Any updates with it? Is there any way to fine-tune generator on some other language?
